readthedocs/api/v3/permissions.py

```diff
@@ -1,31 +1,37 @@
-from rest_framework.permissions import IsAuthenticated
+from rest_framework.permissions import BasePermission
 
 
-class PublicDetailPrivateListing(IsAuthenticated):
+class PublicDetailPrivateListing(BasePermission):
 
     """
     Permission class for our custom use case.
 
     * Always give permission for a ``detail`` request
     * Only give permission for ``listing`` request if user is admin of the project
+    """
+
+    def has_permission(self, request, view):
+        if view.detail:
+            return True
+
+        project = view._get_parent_project()
+        if view.has_admin_permission(request.user, project):
```

Comment:
`has_admin_permission` seems poorly named to me. It isn't immediately clear what it means. I realize it means that the user is a maintainer on the project but maybe it should be more explicit. How about `is_project_maintainer` or something like that. It does seem to mirror the method name on the `ProjectQuerySet` but it is much more clear what that method does in context (`Project.objects.for_admin_user` is fairly clear although could be better while `view.has_admin_permission` is not).

Reply:
`is_project_maintainer` is a good name, but I'm not sure it does fit with corporate (or under an organizational structure) --where only read access is granted.

Even if it doesn't fit 100%, it's still a better name :smile: